### PR TITLE
feat: add df.unstack

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: tests/system/small/test_dataframe.py [hunk 1]
```diff
@@ -1705,6 +1700,26 @@ def test_df_stack(scalars_dfs):
     pd.testing.assert_series_equal(bf_result, pd_result, check_dtype=False)
 
 
+def test_df_unstack(scalars_dfs):
+    scalars_df, scalars_pandas_df = scalars_dfs
+    # To match bigquery dataframes
```
Comment (shobsi) [inline tests/system/small/test_dataframe.py:1705]: Didn't quite get this, why can't we use the original `scalars_pandas_df`?

File: tests/system/small/test_multiindex.py [hunk 2]
```diff
@@ -729,6 +729,26 @@ def test_column_multi_index_stack(scalars_df_index, scalars_pandas_df_index):
     )
 
 
+def test_column_multi_index_unstack(scalars_df_index, scalars_pandas_df_index):
```
Comment (shobsi) [inline tests/system/small/test_multiindex.py:732]: [nit] would `test_multi_column_unstack` better represent this test?

File: tests/system/small/test_multiindex.py [hunk 3]
```diff
@@ -866,6 +886,17 @@ def test_column_multi_index_reorder_levels(scalars_df_index, scalars_pandas_df_i
     pandas.testing.assert_frame_equal(bf_result, pd_result)
 
 
+def test_multi_index_unstack(hockey_df, hockey_pandas_df):
+    bf_result = (
+        hockey_df.set_index(["team_name", "season", "position"]).unstack().to_pandas()
+    )
+    pd_result = hockey_pandas_df.set_index(
+        ["team_name", "season", "position"]
+    ).unstack()
```
Comment (shobsi) [inline tests/system/small/test_multiindex.py:895]: I think it would be worth testing for all levels: `{-3, -2, -1, 0, 1, 2}`

Comment (TrevorBergeron) [reply tests/system/small/test_multiindex.py:895]: I don't support levels param yet. Will add levels support to stack() and unstack() in a later pr.

File: tests/system/small/test_multiindex.py (hunk 2 above)
Comment (TrevorBergeron) [reply tests/system/small/test_multiindex.py:732]: My naming convention is based on the initial dataframe object, column_multi_index indicating that the initial column axis is a multi-index. Open to changing this convention, but it is inline with how I've named the tests so far.

File: tests/system/small/test_dataframe.py (hunk 1 above)
Comment (shobsi) [reply tests/system/small/test_dataframe.py:1705]: So if we remove

```python
scalars_pandas_df = scalars_pandas_df.copy()
```

the following will still apply, just on the original dataframe, no?

```python
scalars_pandas_df.columns = scalars_pandas_df.columns.astype("string[pyarrow]")
```

Comment (TrevorBergeron) [reply tests/system/small/test_dataframe.py:1705]: tests share resources, so we create a copy to make sure our mutation doesn't affect other tests

Comment (shobsi) [reply tests/system/small/test_dataframe.py:1705]: Ah, I was missing that, thanks!

**Review by shobsi — APPROVED** [review]

**Pull request merged**